### PR TITLE
Fixed prod test cases

### PR DESCRIPTION
**Pull request by furqanmlk**
#### Summary

**Fixed following test cases**

```MM-T1477 Accessibility Support in Status Dropdown``` 
```MM-T1702 Search works when changing public/archived options in the dropdown``` 
```MM-T2927_4 should show Status header, with no pointer cursor``` 

**Updated Missing Key**
```SA15008 - Sign In Forgot password - Email address has account on server``` to ```MM-T407```

### CODE REVIEW

**Review by josephbaylon — CHANGES_REQUESTED** [review]
Getting a consistent error on `more_channels_spec`

Screen Shot 2021-03-10 at 4 43 32 PM

File: e2e/cypress/integration/accessibility/accessibility_dropdowns_spec.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ describe('Verify Accessibility Support in Dropdown Menus', () => {
         cy.focused().tab();
 
         // * Verify the accessibility support in the Status Dropdown menu items
-        const labels = ['online', 'away', 'do not disturb. disables all notifications', 'offline'];
+        const labels = [' dialog', 'online', 'away', 'do not disturb. disables all notifications', 'offline'];
```
Comment (saturninoabril) [inline e2e/cypress/integration/accessibility/accessibility_dropdowns_spec.js:122]: Test is now passing but I suggest to correctly fix the accessibility by:
```diff
--- a/components/status_dropdown/status_dropdown.jsx
+++ b/components/status_dropdown/status_dropdown.jsx
@@ -178,7 +178,7 @@ export default class StatusDropdown extends React.PureComponent {
         return (
             <Menu.Group>
                 <Menu.ItemToggleModalRedux
-                    text={''}
+                    accessibilityLabel='Custom Status'
                     modalId={ModalIdentifiers.CUSTOM_STATUS}
```
```diff
--- a/components/widgets/menu/menu_items/menu_item_toggle_modal_redux.tsx
+++ b/components/widgets/menu/menu_items/menu_item_toggle_modal_redux.tsx
@@ -17,15 +17,16 @@ type Props = {
     dialogProps?: Dictionary<any>;
     extraText?: string;
     text: string;
+    accessibilityLabel: string;
     className?: string;
     children?: React.ReactNode;
     sibling?: React.ReactNode;
 }
 
-export const MenuItemToggleModalReduxImpl: React.FC<Props> = ({modalId, dialogType, dialogProps, text, extraText, children, className, sibling}: Props) => (
+export const MenuItemToggleModalReduxImpl: React.FC<Props> = ({modalId, dialogType, dialogProps, text, accessibilityLabel, extraText, children, className, sibling}: Props) => (
     <>
         <ToggleModalButtonRedux
-            accessibilityLabel={text}
+            accessibilityLabel={accessibilityLabel || text}
             modalId={modalId}
```

Then this should be:
```suggestion
        const labels = ['Custom Status dialog', 'online', 'away', 'do not disturb. disables all notifications', 'offline'];
```

Also, please request for Dev review from webapp team for app code changes.

File: e2e/cypress/integration/account_settings/display/theme/custom_theme_sidebar_styles_spec.js [hunk 2]
```diff
@@ -93,24 +93,24 @@ describe('Custom Theme - Sidebar Styles', () => {
         cy.get('#unreadIndicatorBottom').should('have.css', 'color', rgbArrayToString(themeRgbColor.mentionColor));
 
         // # Set user status to online
-        cy.userStatus(0);
+        cy.userStatus(1);
```
Comment (saturninoabril) [inline e2e/cypress/integration/account_settings/display/theme/custom_theme_sidebar_styles_spec.js:96]: While here, I'd like to take the opportunity to improve `cy.userStatus` command by deprecating it totally and replacing with the new command as `cy.uiOpenSetStatusMenu`:
```diff
--- a/e2e/cypress/support/ui/menu.js
+++ b/e2e/cypress/support/ui/menu.js
@@ -3,6 +3,7 @@
 
 const MAIN_MENU = 'main menu';
 const CHANNEL_MENU = 'channel menu';
+const SET_STATUS_MENU = 'set status';
 
 function openMenu(name, item) {
     const menu = () => cy.findByRole('button', {name}).should('be.visible');
@@ -45,3 +46,7 @@ Cypress.Commands.add('uiCloseChannelMenu', () => {
 Cypress.Commands.add('uiGetChannelMenu', () => {
     return getMenu(CHANNEL_MENU);
 });
+
+Cypress.Commands.add('uiOpenSetStatusMenu', (item = '') => {
+    return openMenu(SET_STATUS_MENU, item);
+});
```
Please update `menu.d.ts` as well.

Then this can be:
```suggestion
        cy.uiOpenSetStatusMenu('Online');
```

Other options could be: "Set a Custom Status", "Away", "Do Not Disturb" and "Offline"

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 3]
```diff
@@ -212,6 +212,7 @@ describe('Channels', () => {
                 // * Modal should show the archived channels list
                 cy.wrap(el).should('contain', 'Show: Archived Channels');
             }).wait(TIMEOUTS.HALF_SEC);
+            cy.get('#searchChannelsTextbox').clear();
             cy.get('#moreChannelsList').should('be.visible').children().should('have.length', 2);
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:216]: Failed on me here. Maybe this should be tested in a new team, instead of shared `testTeam`.
Screen Shot 2021-03-11 at 7 21 20 PM

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks Furqan! Please see comments.

**Comment by josephbaylon** [comment]
@furqanmlk Please check build failures

**Comment by furqanmlk** [comment]
> @furqanmlk Please check build failures

Thanks @josephbaylon I am fixing it.
There is a type error in ```menu_item_toggle_modal_redux.test.tsx```

**Comment by saturninoabril** [comment]
@furqanmlk I missed it but you may change the new prop as optional,
```javascript
accessibilityLabel?: string;
```

**Comment by furqanmlk** [comment]
Two Unit Test cases are failing , I am looking at them. 
`components/StatusDropdown should match snapshot with custom status enabled`
and 
`components/StatusDropdown should match snapshot with custom status pulsating dot enabled`

**Comment by saturninoabril** [comment]
Update by `npm run test -- -u` or simply `npm run updatesnapshot`.

File: e2e/cypress/integration/enterprise/profile_popover/profile_popover_spec.js [hunk 4]
```diff
@@ -456,7 +456,7 @@ const verifyAddToChannel = (user, visible = true) => {
 const clickAddToChannel = (user) => {
     // # Open profile popover
     cy.get('#postListContent', {timeout: TIMEOUTS.ONE_MIN}).within(() => {
-        cy.findAllByText(user.username).first().should('have.text', user.username).click();
+        cy.findAllByText(`@${user.username}`).first().should('have.text', '@' + user.username).click();
```
Comment (prapti) [inline e2e/cypress/integration/enterprise/profile_popover/profile_popover_spec.js:459]: Multiple tests are failing for me on this step.

**Review by prapti — COMMENTED** [review]
Thank you for fixing these @furqanmlk! Everything looks good to me, and most tests are passing except for /profile_popover_spec.js where the tests are failing at the exact step you've changed. Is it passing fine for you? Wondering if it's just for me that it's failing.

File: e2e/cypress/integration/enterprise/profile_popover/profile_popover_spec.js (hunk 4 above)
Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/profile_popover/profile_popover_spec.js:459]: Thanks for bringing this up Prapti! All tests passed on me (see screenshot) but upon looking closer, MM-T2927_4 script does not match with step 4 of MM-T2927 (Zephyr). Also, here in code there were 4 steps only while 5 in Zephyr. @furqanmlk please take a look again, thanks!
Screen Shot 2021-03-12 at 5 21 08 PM

Comment (furqanmlk) [reply e2e/cypress/integration/enterprise/profile_popover/profile_popover_spec.js:459]: @saturninoabril thanks for commenting.
Sure I am looking at this, I did not go in the details as it is in prod status. I was just fixing the test cases only.
Dark theme test case is missing.
Should I add the new test case in this PR or should I open new one ? this PR has fixes for other test cases.

File: e2e/cypress/integration/menus/status_dropdown_spec.js [hunk 5]
```diff
@@ -176,9 +176,9 @@ describe('Status dropdown menu', () => {
             // # Open status menu
             cy.get('.MenuWrapper .status-wrapper.status-selector button.status').click();
 
-            // * Verify "Status" header does not have pointer cursor
-            cy.get('.MenuWrapper.status-dropdown-menu .Menu__content.dropdown-menu li:first-child').should('be.visible').
-                and('have.text', 'Status').and('not.have.css', 'cursor', 'pointer');
+            // * Verify "Set a Custom Status" header is clickable
+            cy.get('.MenuWrapper.status-dropdown-menu .Menu__content.dropdown-menu li:nth-child(3)').should('be.visible').
+                and('have.text', 'Set a Custom Status').and('have.css', 'cursor', 'pointer');
```
Comment (saturninoabril) [inline e2e/cypress/integration/menus/status_dropdown_spec.js:181]: @furqanmlk Oh no, sorry I mixed it up. The comment I made at https://github.com/mattermost/mattermost-webapp/pull/7676/files#r593033239 is actually referring to this change.  With that, I'm good to revert this change and address in a separate PR.

File: e2e/cypress/integration/enterprise/profile_popover/profile_popover_spec.js (hunk 4 above)
Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/profile_popover/profile_popover_spec.js:459]: See https://github.com/mattermost/mattermost-webapp/pull/7676/files#r593212532.

**Comment by furqanmlk** [comment]
Thanks @saturninoabril and @prapti 

Fixed more test cases  in ```profile_popover_spec```

image


image

**Review by prapti — APPROVED** [review]
Thank you Furqan! Looks good to me now!

**Review by saturninoabril — APPROVED** [review]
Thanks Furqan! Tested and all passed. LGTM.

**Review by josephbaylon — APPROVED** [review]
LGTM. Just minor change requests
